pysat/instruments/icon_euv.py

```diff
@@ -190,32 +194,28 @@ def clean(inst, clean_level=None):
         Instrument class object, whose attribute clean_level is used to return
         the desired level of data selectivity.
 
-    Returns
-    --------
-    Void : (NoneType)
-        data in inst is modified in-place.
-
     Note
     ----
         Supports 'clean', 'dusty', 'dirty', 'none'. Method is
         not called by pysat if clean_level is None or 'none'.
 
     """
 
-    try:
-        L26_Flag = inst['Flags']
-    except KeyError:
-        L26_Flag = inst['ICON_L26_Flags']
     vars = ['HmF2', 'NmF2', 'Oplus']
-
-    if clean_level == 'clean':
-        idx, = np.where(L26_Flag > 0)
-        inst[idx, vars] = np.nan
-    elif clean_level == 'dusty':
-        idx, = np.where(L26_Flag > 1)
-        inst[idx, vars] = np.nan
-    elif clean_level == 'dirty':
-        idx, = np.where(L26_Flag > 2)
-        inst[idx, vars] = np.nan
+    if 'Flags' in inst.variables:
+        icon_flag = 'Flags'
+    else:
+        icon_flag = 'ICON_L26_Flags'
+        vars = ['ICON_L26_' + x for x in vars]
+
+    if inst.clean_level == 'clean':
+        for var in vars:
+            inst[var] = inst[var].where(inst[icon_flag] == 0)
```

Comment:
The Jupiter notebook released by the ICON team has both 0 and 1 in there for 'good'. I tried loading a few EUV days so far and haven't found any == 0 on the flags variable yet.

Reply:
![EUV_Flags_Example](https://user-images.githubusercontent.com/3595641/87957385-cbf46f80-ca75-11ea-9abc-acac369eeeed.png)

Reply:
The cleaning method does work though :) Thanks Jeff.

Reply:
Wondering if this is a typo.  The other instruments seem to use 1 for good.  Should we update clean and dusty to use 1?